### PR TITLE
Price releaser view

### PR DESCRIPTION
**Pull request by iljahamalainenags**
In this PR:
- added integration with Redux, created slice with reducers
- implemented layout of the page with selectboxes group
- tags and selectboxes are working together via shared state
- selectboxes may have either dynamic content loaded via HTTP request or hardcoded values

Screenshot 2021-03-10 at 11 07 05

### CODE REVIEW

File: src/views/price-releaser/priceReleaseSlice.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            }
+        },
+
+        selectModels: (state, action: PayloadAction<ItemData>) => {
+            if (action.payload.isSelected) {
+                return {
+                    ...state,
+                    models: [...state.models, action.payload]
+                }
+            } else {
```
[the remark this answers is not in the document]
Comment (iljahamalainenags) [reply src/views/price-releaser/priceReleaseSlice.ts:44]: yep, cheers

File: public/tick.icon.svg [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (ianmilliken) [inline public/tick.icon.svg:1]: There are various tick components in react-feather, do we need this file too?

File: src/components/navbar/NavBar.tsx [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+import { useLocation } from "react-router-dom";
 
-export const NavBar = () => (
-    <nav className={styles.navbar}>
+export const NavBar = () => {
+    const location = useLocation();
+    return <nav className={styles.navbar}>
         <ul>
-            <li>
+            <li className={isActive(location.pathname, "/")}>
```
Comment (ianmilliken) [inline src/components/navbar/NavBar.tsx:10]: React Router has the [NavLink](https://reactrouter.com/web/api/NavLink) module to handle active styles, so you shouldn't need your own `isActive()` function

File: src/Router.tsx [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-            </div>
+            <PriceReleaserView />
         </Route>
+
         <Route path="/marketplace-release">
             <div>
-                Marketplace Manager
-
-                <Tag label={"Tag name"} onClick={() => console.log("clicked")}/>
+                Market place release
```
Comment (ianmilliken) [inline src/Router.tsx:14]: Is this a placeholder?

File: src/components/navbar/navbar.module.scss [hunk 5]
```diff
@@ -1,11 +1,49 @@
...
       list-style-type: none;
       float: left;
-      margin: 10px 20px;
+      margin: 20px 20px 0 20px;
+      height: 46px;
+      position: relative;
+
       a {
         text-decoration: none;
+        font-size: 14px;
```
Comment (ianmilliken) [inline src/components/navbar/navbar.module.scss:40]: Just a note on fonts, colors, and other repeating values - we should start putting them into globally scoped vars for easy reuse.

File: src/components/selectbox/selectbox.module.scss [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    cursor: pointer;
+    width: 100%;
+    position: relative;
+    line-height: 50px;
+    height: $input-field-height;
+
+    // preview
+    .selectedSingle {
+
+      .flagIcon IMG {
```
Comment (ianmilliken) [inline src/components/selectbox/selectbox.module.scss:18]: Sorry, nitpicking here but `img` not `IMG`

File: src/components/selectbox/selectbox.module.scss [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+          display: block;
+          height: 18px;
+        }
+      }
+
+      .optionContainer {
+        position: relative;
+        padding: 8px 10px;
+
+        .flagIcon IMG {
```
Comment (ianmilliken) [inline src/components/selectbox/selectbox.module.scss:136]: Same as above

File: src/views/price-releaser/PriceReleaserView.tsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+type GroupProps = {
+    title: string,
+    description: string,
+    url?: string,
+    initialItems?: SelectboxItem[],
+    onChange: (selectedIDs: ItemData)=>void,
+    selected: ItemData[]
+}
+
+const SelectBoxGroup = ({ title, description, url, initialItems, onChange, selected}: GroupProps) => {
```
Comment (ianmilliken) [inline src/views/price-releaser/PriceReleaserView.tsx:108]: Prefer to have one React component per file, please move SelectBoxGroup into its own file

File: src/components/selectbox/Selectbox.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    const [selected, setSelected] = useState<SelectboxIconItem>({} as SelectboxIconItem)
+
+    function change(selected: SelectboxIconItem) {
+        setSelected(selected)
+        if (props.onChange) {
+            props.onChange(selected)
+        }
+    }
+
+    return (<div className={styles.selectbox + " " + props.className}>
```
Comment (ianmilliken) [inline src/components/selectbox/Selectbox.tsx:26]: Prefer template literals strings to concatenated strings for readability, so something like 
```<div className={`${styles.selectbox} ${props.className}`}>```

File: src/components/selectbox/Selectbox.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+type SelectboxProps = {
+    placeholder: string,
+    className?: string,
+    onChange?: (selected: ItemData)=>void,
+    items: SelectboxIconItem[]
+}
+
+export type SelectboxIconItem = ItemData & ItemCallbacks & {icon?: ReactNode};
+
+export const Selectbox: FunctionComponent<SelectboxProps> = (props) => {
```
Comment (ianmilliken) [inline src/components/selectbox/Selectbox.tsx:15]: Please destructure props

**Review by ianmilliken — CHANGES_REQUESTED** [review]
Minor changes

File: src/components/navbar/NavBar.tsx (hunk 3 above)
Comment (iljahamalainenags) [reply src/components/navbar/NavBar.tsx:10]: ahhh :))) cheers, didn't know that

File: public/tick.icon.svg (hunk 2 above)
Comment (iljahamalainenags) [reply public/tick.icon.svg]: replaced, thanks!

File: src/Router.tsx (hunk 4 above)
Comment (iljahamalainenags) [reply src/Router.tsx:14]: yes, this page is not ready :)

File: src/components/selectbox/Selectbox.tsx (hunk 9 above)
Comment (iljahamalainenags) [reply src/components/selectbox/Selectbox.tsx:26]: nice one! cheers!

**Review by ianmilliken — APPROVED** [review]

**Review by carmo-evan — APPROVED** [review]

**Pull request merged**